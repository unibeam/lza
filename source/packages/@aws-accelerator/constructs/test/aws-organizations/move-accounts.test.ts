/**
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

import * as cdk from 'aws-cdk-lib';
import { MoveAccounts } from '../../index';
import { snapShotTest } from '../snapshot-test';

const testNamePrefix = 'Construct(MoveAccounts): ';

//Initialize stack for snapshot test and resource configuration test
const stack = new cdk.Stack();

new MoveAccounts(stack, 'MoveAccounts', {
  globalRegion: 'us-west-2',
  commitId: 'sample-commit-id',
  managementAccountId: stack.account,
  configTable: new cdk.aws_dynamodb.Table(stack, 'CustomTable', {
    partitionKey: { name: 'dataType', type: cdk.aws_dynamodb.AttributeType.STRING },
  }),
  lambdaKmsKey: new cdk.aws_kms.Key(stack, 'CustomLambdaKey', {}),
  cloudWatchLogsKmsKey: new cdk.aws_kms.Key(stack, 'CustomCWLKey', {}),
  cloudWatchLogRetentionInDays: 365,
  controlTower: false,
});
/**
 * MoveAccounts construct test
 */
describe('MoveAccounts', () => {
  snapShotTest(testNamePrefix, stack);
});
