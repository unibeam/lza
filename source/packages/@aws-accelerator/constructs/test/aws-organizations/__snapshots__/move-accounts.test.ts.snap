// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoveAccounts Construct(MoveAccounts):  Snapshot Test 1`] = `
{
  "Mappings": {
    "LatestNodeRuntimeMap": {
      "af-south-1": {
        "value": "nodejs20.x",
      },
      "ap-east-1": {
        "value": "nodejs20.x",
      },
      "ap-northeast-1": {
        "value": "nodejs20.x",
      },
      "ap-northeast-2": {
        "value": "nodejs20.x",
      },
      "ap-northeast-3": {
        "value": "nodejs20.x",
      },
      "ap-south-1": {
        "value": "nodejs20.x",
      },
      "ap-south-2": {
        "value": "nodejs20.x",
      },
      "ap-southeast-1": {
        "value": "nodejs20.x",
      },
      "ap-southeast-2": {
        "value": "nodejs20.x",
      },
      "ap-southeast-3": {
        "value": "nodejs20.x",
      },
      "ap-southeast-4": {
        "value": "nodejs20.x",
      },
      "ap-southeast-5": {
        "value": "nodejs20.x",
      },
      "ap-southeast-7": {
        "value": "nodejs20.x",
      },
      "ca-central-1": {
        "value": "nodejs20.x",
      },
      "ca-west-1": {
        "value": "nodejs20.x",
      },
      "cn-north-1": {
        "value": "nodejs18.x",
      },
      "cn-northwest-1": {
        "value": "nodejs18.x",
      },
      "eu-central-1": {
        "value": "nodejs20.x",
      },
      "eu-central-2": {
        "value": "nodejs20.x",
      },
      "eu-isoe-west-1": {
        "value": "nodejs18.x",
      },
      "eu-north-1": {
        "value": "nodejs20.x",
      },
      "eu-south-1": {
        "value": "nodejs20.x",
      },
      "eu-south-2": {
        "value": "nodejs20.x",
      },
      "eu-west-1": {
        "value": "nodejs20.x",
      },
      "eu-west-2": {
        "value": "nodejs20.x",
      },
      "eu-west-3": {
        "value": "nodejs20.x",
      },
      "il-central-1": {
        "value": "nodejs20.x",
      },
      "me-central-1": {
        "value": "nodejs20.x",
      },
      "me-south-1": {
        "value": "nodejs20.x",
      },
      "mx-central-1": {
        "value": "nodejs20.x",
      },
      "sa-east-1": {
        "value": "nodejs20.x",
      },
      "us-east-1": {
        "value": "nodejs20.x",
      },
      "us-east-2": {
        "value": "nodejs20.x",
      },
      "us-gov-east-1": {
        "value": "nodejs18.x",
      },
      "us-gov-west-1": {
        "value": "nodejs18.x",
      },
      "us-iso-east-1": {
        "value": "nodejs18.x",
      },
      "us-iso-west-1": {
        "value": "nodejs18.x",
      },
      "us-isob-east-1": {
        "value": "nodejs18.x",
      },
      "us-west-1": {
        "value": "nodejs20.x",
      },
      "us-west-2": {
        "value": "nodejs20.x",
      },
    },
  },
  "Resources": {
    "CustomCWLKey7119CF89": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "KeyPolicy": {
          "Statement": [
            {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
    "CustomLambdaKey2287F5A9": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "KeyPolicy": {
          "Statement": [
            {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
    "CustomTable90F1C476": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AttributeDefinitions": [
          {
            "AttributeName": "dataType",
            "AttributeType": "S",
          },
        ],
        "KeySchema": [
          {
            "AttributeName": "dataType",
            "KeyType": "HASH",
          },
        ],
        "ProvisionedThroughput": {
          "ReadCapacityUnits": 5,
          "WriteCapacityUnits": 5,
        },
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Retain",
    },
    "MoveAccounts99C70985": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "MoveAccountsMoveAccountsFunctionLogGroupFAEDB078",
      ],
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "MoveAccountsMoveAccountsProviderframeworkonEvent5D4E26AA",
            "Arn",
          ],
        },
        "commitId": "sample-commit-id",
        "configTableName": {
          "Ref": "CustomTable90F1C476",
        },
        "globalRegion": "us-west-2",
        "stackName": "Default",
        "uuid": "REPLACED-UUID",
      },
      "Type": "Custom::MoveAccounts",
      "UpdateReplacePolicy": "Delete",
    },
    "MoveAccountsMoveAccountsFunctionD31A3773": {
      "DependsOn": [
        "MoveAccountsMoveAccountsFunctionServiceRoleDefaultPolicyB573AE3E",
        "MoveAccountsMoveAccountsFunctionServiceRoleADDE3D6B",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "REPLACED-GENERATED-NAME.zip",
        },
        "Description": "Moves accounts to conform account config",
        "Handler": "index.handler",
        "KmsKeyArn": {
          "Fn::GetAtt": [
            "CustomLambdaKey2287F5A9",
            "Arn",
          ],
        },
        "Role": {
          "Fn::GetAtt": [
            "MoveAccountsMoveAccountsFunctionServiceRoleADDE3D6B",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "MoveAccountsMoveAccountsFunctionLogGroupFAEDB078": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "KmsKeyId": {
          "Fn::GetAtt": [
            "CustomCWLKey7119CF89",
            "Arn",
          ],
        },
        "LogGroupName": {
          "Fn::Join": [
            "",
            [
              "/aws/lambda/",
              {
                "Ref": "MoveAccountsMoveAccountsFunctionD31A3773",
              },
            ],
          ],
        },
        "RetentionInDays": 365,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "MoveAccountsMoveAccountsFunctionServiceRoleADDE3D6B": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "MoveAccountsMoveAccountsFunctionServiceRoleDefaultPolicyB573AE3E": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "organizations:ListAccountsForParent",
                "organizations:ListRoots",
                "organizations:ListOrganizationalUnitsForParent",
                "organizations:MoveAccount",
              ],
              "Effect": "Allow",
              "Resource": "*",
              "Sid": "OrganizationsAccess",
            },
            {
              "Action": "dynamodb:Query",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "CustomTable90F1C476",
                  "Arn",
                ],
              },
              "Sid": "DynamodbTableAccess",
            },
            {
              "Action": "cloudformation:DescribeStacks",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":cloudformation:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":stack/Default*",
                  ],
                ],
              },
              "Sid": "CloudformationAccess",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "MoveAccountsMoveAccountsFunctionServiceRoleDefaultPolicyB573AE3E",
        "Roles": [
          {
            "Ref": "MoveAccountsMoveAccountsFunctionServiceRoleADDE3D6B",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "MoveAccountsMoveAccountsProviderframeworkonEvent5D4E26AA": {
      "DependsOn": [
        "MoveAccountsMoveAccountsProviderframeworkonEventServiceRoleDefaultPolicy04895FAA",
        "MoveAccountsMoveAccountsProviderframeworkonEventServiceRoleB3998377",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "REPLACED-GENERATED-NAME.zip",
        },
        "Description": "AWS CDK resource provider framework - onEvent (Default/MoveAccounts/MoveAccountsProvider)",
        "Environment": {
          "Variables": {
            "USER_ON_EVENT_FUNCTION_ARN": {
              "Fn::GetAtt": [
                "MoveAccountsMoveAccountsFunctionD31A3773",
                "Arn",
              ],
            },
          },
        },
        "Handler": "framework.onEvent",
        "Role": {
          "Fn::GetAtt": [
            "MoveAccountsMoveAccountsProviderframeworkonEventServiceRoleB3998377",
            "Arn",
          ],
        },
        "Runtime": {
          "Fn::FindInMap": [
            "LatestNodeRuntimeMap",
            {
              "Ref": "AWS::Region",
            },
            "value",
          ],
        },
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "MoveAccountsMoveAccountsProviderframeworkonEventServiceRoleB3998377": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "MoveAccountsMoveAccountsProviderframeworkonEventServiceRoleDefaultPolicy04895FAA": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "MoveAccountsMoveAccountsFunctionD31A3773",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "MoveAccountsMoveAccountsFunctionD31A3773",
                          "Arn",
                        ],
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "MoveAccountsMoveAccountsProviderframeworkonEventServiceRoleDefaultPolicy04895FAA",
        "Roles": [
          {
            "Ref": "MoveAccountsMoveAccountsProviderframeworkonEventServiceRoleB3998377",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
